ConcurrentModificationException in HttpSignProvider

## Environment Details
* Helidon Version: 2.5.4
* Helidon SE
* JDK version: 11
* OS: Linux

----------

## Problem Description

Under load, following exception is encountered in `HttpSignProvider`:

```
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap.computeIfAbsent(HashMap.java:1135) ~[?:?]
	at io.helidon.security.providers.httpsign.HttpSignProvider.lambda$signRequest$9(HttpSignProvider.java:270) ~[helidon-security-providers-http-sign-2.5.4.jar:2.5.4]
	at java.util.Optional.map(Optional.java:265) ~[?:?]
	at io.helidon.security.providers.httpsign.HttpSignProvider.signRequest(HttpSignProvider.java:269) ~[helidon-security-providers-http-sign-2.5.4.jar:2.5.4]
	at io.helidon.security.providers.httpsign.HttpSignProvider.lambda$outboundSecurity$4(HttpSignProvider.java:261) ~[helidon-security-providers-http-sign-2.5.4.jar:2.5.4]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) ~[?:?]
	at io.helidon.common.context.Contexts.runInContext(Contexts.java:117) ~[helidon-common-context-2.5.4.jar:2.5.4]
	at io.helidon.common.context.ContextAwareExecutorImpl.lambda$wrap$7(ContextAwareExecutorImpl.java:154) ~[helidon-common-context-2.5.4.jar:2.5.4]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	... 1 more
```

In order to minimize latency of our reverse proxy, we cache/reuse HttpSignProvider across requests. Is that valid?

--- a/security/providers/http-sign/src/main/java/io/helidon/security/providers/httpsign/HttpSignProvider.java
+++ b/security/providers/http-sign/src/main/java/io/helidon/security/providers/httpsign/HttpSignProvider.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018, 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2018, 2023 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/security/providers/http-sign/src/main/java/io/helidon/security/providers/httpsign/HttpSignProvider.java
+++ b/security/providers/http-sign/src/main/java/io/helidon/security/providers/httpsign/HttpSignProvider.java
@@ -25,6 +25,7 @@ import java.util.Set;
 import java.util.TreeMap;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
 
 import io.helidon.config.Config;
 import io.helidon.security.AuthenticationResponse;
@@ -76,7 +77,7 @@ public final class HttpSignProvider implements AuthenticationProvider, OutboundS
     private final Map<String, InboundClientDefinition> inboundKeys;
     private final OutboundConfig outboundConfig;
     // cache of target name to a signature configuration for outbound calls
-    private final Map<String, OutboundTargetDefinition> targetKeys = new HashMap<>();
+    private final Map<String, OutboundTargetDefinition> targetKeys = new ConcurrentHashMap<>();
     private final boolean backwardCompatibleEol;
 
     private HttpSignProvider(Builder builder) {

